include/daemonize.hpp

```diff
@@ -0,0 +1,24 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0. This product includes software
+// developed at Datadog (https://www.datadoghq.com/). Copyright 2021-Present
+// Datadog, Inc.
+
+#pragma once
+
+#include <functional>
+#include <sys/types.h>
+
+namespace ddprof {
+
+struct DaemonizeResult {
```

Comment:
This API looks nice considering what it was in the beginning 👍 . I had trouble figuring out how we could design an interface around this